app/model/sdl/Abstract/AppModel.js

```diff
@@ -622,15 +622,15 @@ SDL.ABSAppModel = Em.Object.extend(
             };
             // Insert new item at calculated position
             commands.splice(position, 0, newItem);
-        		if (SDL.SDLController.getApplicationModel(request.params.appID) &&
+        	if (SDL.SDLController.getApplicationModel(request.params.appID) &&
                     SDL.OptionsView.active) {
-                        SDL.SDLController.buttonsSort(parentID, this.appID);
-                        SDL.OptionsView.commands.refreshItems();
-        		}
+              SDL.SDLController.buttonsSort(parentID, this.appID);
+              SDL.OptionsView.commands.refreshItems();
+        	}
 
-    		    if(request.params.cmdIcon) {
+    		if(request.params.cmdIcon || request.params.secondaryImage) {
 
-              var image = request.params.cmdIcon;
+              var image = Object.assign({}, request.params.cmdIcon, request.params.secondaryImage);
```

Comment:
@LitvinenkoIra I don't think this would work properly, especially if request contains both images.
Note, that `validateImages()` accepts the array of image objects, so better approach would be like that:
```
var imageList = [];
if (request.params.cmdIcon) {
  imageList.push(request.params.cmdIcon);
}

if (request.params.secondaryImage) {
  imageList.push(request.params.secondaryImage);
}

if (imageList.length > 0) {
 var callback = function(failed, info) {
                var WARNINGS = SDL.SDLModel.data.resultCode.WARNINGS;
                var SUCCESS = SDL.SDLModel.data.resultCode.SUCCESS;
                FFW.UI.sendUIResult(
                  failed ? WARNINGS : SUCCESS, 
                  request.id, 
                  request.method, 
                  info);
              }
  SDL.SDLModel.validateImages(request.id, callback, imageList);
  return; 
}
```

Reply:
Thank you! Done in 6c142a743f31cc688e4924355148c10a710d3246